kDrive/UI/Controller/Files/FileActionsFloatingPanelViewController+Actions.swift

```diff
@@ -73,7 +73,7 @@ extension FileActionsFloatingPanelViewController {
             case .manageDropbox:
                 return file.isDropbox
             case .folderColor:
-                return file.capabilities.canColor
```

Comment:
suggestion: I think we should have a more complex condition here.
If the user has a free pack, then we can show the option if `file.isDirectory`, else we should listen to `file.capabilities.canColor`.

Reply:
I hear you, but I think you over complexity things.
This logic enables the action in the view if it is a directory, it does not check the capabilities.
Then performing the action, a capability check is performed. And we either display an up-sale sheet if the capability is not there, or we perform the action. Returning false here will hide the action. The way it was built before me is this one I believe.

Reply:
Maybe I could rename the Action to explicit that it will check capabilities on its own.

Reply:
You can also check in `MultipleSelectionFloatingPanelViewController` where a similar logic is performed, there is zero check on capability.